docs/components/modeler/feel/builtin-functions/feel-built-in-functions-miscellaneous.md

```diff
@@ -0,0 +1,157 @@
+---
+id: feel-built-in-functions-miscellaneous
+title: Miscellaneous functions
+description: "This document outlines miscellaneous functions and a few examples."
+---
+
+import MarkerCamundaExtension from "@site/src/mdx/MarkerCamundaExtension";
+
+## fromAi(value)
+
+<MarkerCamundaExtension></MarkerCamundaExtension>
+
+Returns the unmodified `value` parameter. The purpose of this function is solely to tag the value as being generated by
+an AI integration. The actual handling is not done by the FEEL engine, but by a custom integration such as a connector
+or a job worker.
+
+See the following function overloads for additional parameters to the function.
+
+**Function signature**
+
+```feel
+fromAi(value: Any): Any
+```
+
+**Examples**
+
+```feel
+fromAi(toolCall.searchQuery)
+// toolCall.searchQuery contents
+
+fromAi(toolCall.userId)
+// toolCall.userId contents
+
+fromAi("A")
+// "A" - does not make much sense and might be flagged as error by the integration consumiong the definition
+```
+
+## fromAi(value, description)
+
+<MarkerCamundaExtension></MarkerCamundaExtension>
+
+Returns the unmodified `value` parameter. In addition to the previous overload, it also accepts an optional
+`description`
+parameter that can be used to provide a textual description of the value.
+
+**Function signature**
+
+```feel
+fromAi(value: Any, description: string): Any
+```
+
+**Examples**
+
+```feel
+fromAi(toolCall.searchQuery, "The search query used to find the best match.")
+// toolCall.searchQuery contents
+
+fromAi(toolCall.searchQuery, null)
+// toolCall.searchQuery contents
+```
+
+## fromAi(value, description, type)
+
+<MarkerCamundaExtension></MarkerCamundaExtension>
+
+Returns the unmodified `value` parameter. In addition to the previous overload, it also accepts an optional `type`
+parameter that can be used to provide type information about the value.
+
+**Function signature**
+
+```feel
+fromAi(value: Any, description: string, type: string): Any
+```
+
+**Examples**
+
+```feel
+fromAi(toolCall.searchQuery, "The search query used to find the best match.", "string")
+// toolCall.searchQuery contents
+
+fromAi(toolCall.userId, "The user's ID", "number")
+// toolCall.userId contents
+
+fromAi(toolCall.userId, null, "number")
+// toolCall.userId contents
+
+fromAi(value: toolCall.userId, type: "number")
+// toolCall.userId contents
+```
+
+## fromAi(value, description, type, schema)
+
+<MarkerCamundaExtension></MarkerCamundaExtension>
+
+Returns the unmodified `value` parameter. In addition to the previous overload, it also accepts an optional `schema`
+parameter that can be used to provide a (partial) [JSON schema](https://json-schema.org/) for the value.
+
+Note that the schema is not validated by the FEEL engine but might be by a custom integration consuming the information.
+Also, from the engine side it is possible to specify both a `type` and a `schema`, and it depends on the integration on
+which value takes precedence.
```

Comment:
:thought_balloon: Eventually, we should document the usage as part of the agentic-AI page (ad-hoc sub-process). The logic is outside of the FEEL engine and depends on the AI integration.